packages/arco-lib/src/widgets/IconWidget.tsx

```diff
@@ -0,0 +1,183 @@
+};
+type IconStyle = keyof typeof iconStyle;
+
+export const PopoverContent: React.FC<{
+  onChange: (iconName: string) => void;
+  closePopover: () => void;
+}> = ({ onChange, closePopover }) => {
+  const [style, setStyle] = useState<IconStyle>('outline');
+  const [searchValue, setSearchValue] = useState('');
+
+  const categories = useMemo(() => {
+    return mapValues(IconCategories, o => {
+      return mapValues(o, iconComps => {
+        return iconComps.filter(iconComp =>
+          new RegExp(searchValue, 'i').test(iconComp.componentName)
+        );
+      });
+    });
+  }, [searchValue]);
+
+  return (
+    <div>
+      <Space direction="horizontal">
+        <Radio.Group
+          size="small"
+          type="button"
+          value={style}
+          onChange={style => {
+            setStyle(style);
+          }}
+        >
+          {Object.keys(iconStyle).map(key => (
+            <Radio key={key} value={key}>
+              {iconStyle[key as IconStyle]}
+            </Radio>
+          ))}
+        </Radio.Group>
+        <Input
+          value={searchValue}
+          onChange={v => {
+            setSearchValue(v);
+          }}
+          placeholder="Search icon by name"
+          size="small"
+        />
+      </Space>
+      <div className={iconListWrapperStyle}>
+        {Object.keys(categories).map(category => {
+          const curCategory = categories[category as keyof typeof categories];
+          const currentIcons = curCategory[style as keyof typeof curCategory];
+
+          return (
+            <div key={category}>
+              {currentIcons && currentIcons.length > 0 && (
+                <div>
+                  <div className="title">{eng[category as keyof typeof categories]}</div>
+                  <div className="content">
+                    {currentIcons.map(component => {
+                      const iconName = component.componentName;
+                      const Icon = Icons[iconName as keyof typeof Icons];
+                      return (
+                        <Button
+                          onClick={() => {
+                            onChange(iconName);
+                            closePopover();
+                          }}
+                          title={iconName}
+                          key={iconName}
+                          icon={<Icon />}
+                          className={iconButtonStyle}
+                        />
+                      );
+                    })}
+                  </div>
+                </div>
+              )}
+            </div>
+          );
+        })}
+      </div>
+    </div>
+  );
+};
+
+const _IconWidget: React.FC<WidgetProps<IconWidgetWidgetID, string>> = props => {
+  const { value, onChange } = props;
+  const CurrentIcon = Icons[value as keyof typeof Icons];
+  const [visible, setVisible] = useState(false);
+
+  const closePopover = useCallback(() => {
+    setVisible(false);
+  }, []);
+
+  return (
+    <Popover
```

Comment:
Assign a `containerRef` to `Popover`, as `PopoverWidget` did. Otherwise, it will append popover content to body and cause more rendering.

Reply:
done